core/src/main/java/com/linecorp/armeria/client/HttpClientFactory.java

```diff
@@ -468,4 +468,17 @@ HttpChannelPool pool(EventLoop eventLoop) {
                                                               sslCtxHttp1Or2, sslCtxHttp1Only,
                                                               connectionPoolListener()));
     }
+
+    void tls(TlsKeyPair tlsKeyPair) {
+        final Consumer<? super SslContextBuilder> tlsCustomizer =
+                    customizer -> customizer.keyManager(tlsKeyPair.privateKey(), tlsKeyPair.keyCertChain());
+        final ImmutableList<? extends Consumer<? super SslContextBuilder>> tlsCustomizers =
+                    ImmutableList.of(tlsCustomizer);
+
+        final boolean tlsAllowUnsafeCiphers = options.tlsAllowUnsafeCiphers();
+        sslCtxHttp1Or2 = SslContextUtil
+                .createSslContext(SslContextBuilder::forClient, false, tlsAllowUnsafeCiphers, tlsCustomizers);
+        sslCtxHttp1Only = SslContextUtil
+                .createSslContext(SslContextBuilder::forClient, true, tlsAllowUnsafeCiphers, tlsCustomizers);
```

Comment:
This is not thread-safe. A `ClientFactory` can be accessed by multiple threads. The values set might not be different when it is actually used.